Generation/src/components/PythiaInterface.cpp

```diff
@@ -32,17 +35,25 @@ StatusCode PythiaInterface::initialize() {
   if (System::getEnv("PYTHIA8_XML") != "UNKNOWN") xmlpath = System::getEnv("PYTHIA8_XML") ;
 
   // Initialize pythia
-  m_pythia = std::unique_ptr<Pythia8::Pythia>(new Pythia8::Pythia(xmlpath));
-
-  // Read Pythia configuration file
-  m_pythia->readFile( m_parfile.c_str() );
+  m_pythiaSignal = std::unique_ptr<Pythia8::Pythia>(new Pythia8::Pythia(xmlpath));
+  m_pythiaPileup = std::unique_ptr<Pythia8::Pythia>(new Pythia8::Pythia(xmlpath));
+
+  // Read Pythia configuration files
+  m_pythiaSignal->readFile( m_parfile.c_str() );
+  // do not bother with pileup configuration if no pileup
+  if (0. < m_pileUpTool->getMeanPileUp()) {
+    if (m_pileUpTool->getFilename().empty()) {
+     return Error ( "Define Pythia8 configuration file for pileup in pileuptool (*.cmd)!" );  
+    }
+    m_pythiaSignal->readFile(m_pileUpTool->getFilename().c_str());
+  }
```

Comment:
Sorry, I mis-read the above pointer declaration. Now I see that we actually do have 2 pythia instances. 

So what we should have is a specialised PileUp tool for generating PU with Pythia. That tool then owns the corresponding Pythia instance and then we can move this there as well.